airflow/providers/amazon/aws/utils/waiter_with_logging.py

```diff
@@ -31,10 +33,10 @@ def wait(
     waiter: Waiter,
     waiter_delay: int,
     max_attempts: int,
-    args: dict,
+    args: dict[str, Any],
```

Comment:
the method was introduced like yesterday, so no trouble on that front, the only users are Syed and I for now :p

Reply:
(also, since python is not type-checked, type annotations are only used to produce warnings in static checkers like mypy & such, but changing type annotations will never break anything at runtime)

Reply:
Sounds good :)